Synchronize WPT cookiestore

annevk opened the pull request:
#### b7e1d1a3e53887653a7cd0e4ce079945184f6a6a
<pre>
Synchronize WPT cookiestore
<a href="https://bugs.webkit.org/show_bug.cgi?id=296616">https://bugs.webkit.org/show_bug.cgi?id=296616</a>

Reviewed by Rupin Mittal.

The directory got renamed as part of the specification moving into the
WHATWG. Relevant upstream commit:

<a href="https://github.com/web-platform-tests/wpt/commit/cfd9285284893e6d63d7770deae0789d7f7457d4">https://github.com/web-platform-tests/wpt/commit/cfd9285284893e6d63d7770deae0789d7f7457d4</a>
Canonical link: <a href="https://commits.webkit.org/298163@main">https://commits.webkit.org/298163@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bb2f2cc116914fe100362be7c80c247fddf48203

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114368 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34113 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24577 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120535 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65090 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/28ed2d72-a498-48e2-99cb-5671602f5ed9) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34744 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42674 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86913 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/41827 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/3814ec11-c883-4dba-ab10-3f0b2d9c2478) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117316 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27672 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102715 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67305 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26853 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20841 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64222 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97047 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20956 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/123744 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41382 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30863 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95737 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41759 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98912 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95521 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40672 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18512 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/37425 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18338 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41262 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40856 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44163 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42605 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

RupinMittal commented on LayoutTests/TestExpectations:
```diff
@@ -602,11 +602,25 @@ imported/w3c/web-platform-tests/x-frame-options/get-decode-split.html [ DumpJSCo
+imported/w3c/web-platform-tests/cookiestore/serviceworker_cookiechange_eventhandler_no_change.https.any.serviceworker.html [ Skip ]
+imported/w3c/web-platform-tests/cookiestore/serviceworker_cookiechange_eventhandler_overlapping_subscriptions.https.any.serviceworker.html [ Skip ]
+imported/w3c/web-platform-tests/cookiestore/serviceworker_cookiechange_eventhandler_single_subscription.https.any.serviceworker.html [ Skip ]
+imported/w3c/web-platform-tests/cookiestore/serviceworker_oncookiechange_eventhandler_single_subscription.https.any.serviceworker.html [ Skip ]
```
Yep, skipping the tests that are timing out is the right move. But for all the `serviceworker_cookiechange_eventhandler` ones that we're skipping, let's put them in their own block with a comment at the top saying `# Cookie Store API change events are not supported for service workers`

RupinMittal replied:
As for the failures on stress and on the other bots--are these results on EWS different than what you're seeing locally?

annevk replied:
Yes, and while I fixed some of those some are remaining and I suspect those are intermittent. We'll have to skip those as well or mark them PASS/FAIL. 🙁

RupinMittal replied:
Ah it looks like some of these failures have to do with partitioned cookies. Some partitioned cookie tests have been added to already existing files with tests. We don't support partitioned yet. But unfortunately I don't think we have a way of skipping subtests so let's do this:

There are 4 failing tests on the one failing bot:

`cookiestore/change_eventhandler_for_already_expired.https.window.html `
and
`change_eventhandler_for_no_change.https.window.html`

SKIP since these two have do with partitioned cookies (mention partitioned in the comment)

`change_eventhandler_for_document_cookie.https.window.html`
and 
`imported/w3c/web-platform-tests/cookiestore/change_eventhandler_for_http_cookie_and_set_cookie_headers.https.window.html`

These two have nothing to do with partitioned, so I guess they're just flaky. It would be nice if you could mark these as pass/fail with a bug/radar in the comment (You can assign the radars to me).

Thank you! Hopefully this makes the bots green.

RupinMittal reviewed (commented):
Skipping the timeouts is a good idea. Looks like some tests expectations need to be adjusted as well given the failures on the bots.

RupinMittal reviewed (approved):
LGTM

annevk committed:
Synchronize WPT cookiestore
https://bugs.webkit.org/show_bug.cgi?id=296616

Reviewed by Rupin Mittal.

The directory got renamed as part of the specification moving into the
WHATWG. Relevant upstream commit:

https://github.com/web-platform-tests/wpt/commit/cfd9285284893e6d63d7770deae0789d7f7457d4
Canonical link: https://commits.webkit.org/298163@main
